src/ui/components/TransferGrain.js

```diff
@@ -18,19 +20,74 @@ export type Props = {|
   +setLedger: (Ledger) => void,
 |};
 
+const useStyles = makeStyles({
+  root: {
+    width: "1100px",
+    margin: "0 auto",
+    background: "white",
+    padding: "0 5em 5em",
+  },
+  arrowBody: {
+    width: "300px",
+    marginLeft: "20px",
+    background: "#C9CED2",
+    padding: "5px 20px",
+  },
+  triangle: {
+    width: 0,
+    height: 0,
+    background: "white",
+    borderTop: "30px solid transparent",
+    borderBottom: "30px solid transparent",
+    borderLeft: "30px solid #C9CED2",
+  },
+  dropdown: {
+    width: "300px",
+    height: "50px",
+    border: "1px solid #9EADBA",
+    borderRadius: "500px",
+    background: "#F2F5F7",
+  },
+  centerRow: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  third: {
+    width: "300px",
+    height: "50px",
+    margin: "30px 15px",
+  },
+  half: {
+    width: "450px",
+    height: "50px",
+    margin: "30px",
+  },
+  memo: {
+    width: "960px",
+    height: "50px",
+    border: "1px solid #9EADBA",
+    paddingLeft: "20px",
+    color: "black",
+  },
+});
+
 export const TransferGrain = ({ledger, setLedger}: Props) => {
+  const classes = useStyles();
   const [sourceIdentity, setSourceIdentity] = useState<Identity | null>(null);
+  const [sourceBalance, setSourceBalance] = useState<Grain>(fromInteger(0));
   const [destIdentity, setDestIdentity] = useState<Identity | null>(null);
+  const [destBalance, setDestBalance] = useState<Grain>(fromInteger(0));
   const [amount, setAmount] = useState<string>("0");
-  const [maxAmount, setMaxAmount] = useState<Grain>(fromInteger(0));
   const [memo, setMemo] = useState<string>("");
 
   const setSender = (acct: Account | null) => {
-    setMaxAmount(acct ? acct.balance : fromInteger(0));
+    setSourceBalance(acct ? acct.balance : fromInteger(0));
     setSourceIdentity(acct ? acct.identity : null);
```

Comment:
So we set the balance and the identity in separate calls -- why not just set the account directly, since it has both?

Reply:
sure, done